LibVLCSharp/Shared/Media.cs

```diff
@@ -704,7 +710,8 @@ static void RemoveStream(IntPtr handle)
             DicStreams.TryRemove(handle, out var result);
         }
 
-        void Retain()
+        /// <summary>Increments the native reference counter for the media</summary>
+        public void Retain()
```

Comment:
> How could one use that properly? How do you "release" something that you retained?

You'd release it by calling dispose though it doesn't support it right now.

> I don't see any good use case since we are in a managed world, one could create his own refcount mechanism before disposing

Sure. I implemented it for the sake of completeness more than anything. I'll set it to `internal`

Reply:
Is it used somewhere? If not, I'd remove it and put it in `NotGonnaImplementYetUntilActualUserDemandAndIfItMakesSense` 😄

Reply:
Not yet, but a similar concept was used in https://github.com/videolan/libvlcsharp/blob/5160c3308eb7b86e26cd9fd661fd02fc429f1212/LibVLCSharp/Shared/RendererDiscoverer.cs#L151 and they use the `retain` methods in vlckit so we probably might need it in the future. I'll keep it for now.